### ISSUE TITLE
Self-hosted Grist doesn't work

### ISSUE THREAD

**Issue by KucharczykL** [gristlabs/grist-core#156]
Hello, I followed the example in the README to get Grist up and running and created a container exposed on a port as described in the README:

```
To run Grist running on your computer with [Docker](https://www.docker.com/get-started), do:

docker pull gristlabs/grist
docker run -p 8484:8484 -it gristlabs/grist

Then visit http://localhost:8484 in your browser. You'll be able to create, edit, import, and export documents.
```

However, when I accessed that URL from the browser, I get an error:

![image](https://user-images.githubusercontent.com/31072879/155997202-dd1d2f7e-44fc-43f1-8176-e91cb5961ce6.png)

I then tried clicking the **Go to main page** button and the **Sign in** button but none of them worked.

Logs:
```
yarn run v1.22.17
$ NODE_PATH=_build:_build/stubs node _build/stubs/app/server/server.js
Welcome to Grist.
2022-02-28 14:10:37.251 - info: == Grist version is 0.1.1 (commit unknown)
2022-02-28 14:10:37.259 - info: Loading empty config because /persist/config.json missing
2022-02-28 14:10:37.404 - warn: did not find an appropriately named example workspace in deployment
2022-02-28 14:10:37.406 - info: No plugins directory: ENOENT: no such file or directory, scandir '/.grist/plugins'
2022-02-28 14:10:37.411 - info: Found 1 valid plugins on the system
2022-02-28 14:10:37.411 - debug: PLUGIN builtIn/core -- /plugins/core
2022-02-28 14:10:37.415 - info: Server timeouts: keepAliveTimeout 305000 headersTimeout 306000
2022-02-28 14:10:37.417 - info: server(home,docs,static) available at 0.0.0.0:8484
2022-02-28 14:10:37.430 - info: activity docCount=0, orgCount=0, orgInGoodStandingCount=0, userCount=4, userWithLoginCount=4
2022-02-28 14:10:37.486 - warn: Failed to create GoogleAuth endpoint: GOOGLE_CLIENT_SECRET is not defined
2022-02-28 14:10:37.494 - info: == appRoot: /
2022-02-28 14:10:37.495 - info: == docsRoot: /persist/docs
2022-02-28 14:10:37.495 - info: == defaultBaseDomain: undefined
2022-02-28 14:10:37.495 - info: == pluginUrl: undefined
2022-02-28 14:10:37.495 - info: == instanceRoot: /persist
2022-02-28 14:10:37.495 - info: == tag: unknown
2022-02-28 14:10:37.496 - info: == database: sqlite:///persist/home.sqlite3
2022-02-28 14:10:37.496 - info: == userRoot: /.grist
2022-02-28 14:10:37.496 - info: == loginMiddlewareComment: no-logins
2022-02-28 14:10:37.496 - info: == s3: disabled
2022-02-28 14:10:37.497 - info: == docWorkerId: testDocWorkerId_8484
2022-02-28 14:10:48.827 - debug: Auth[GET]: id 1 email anon@getgrist.com host 192.168.0.106:8888 path / org 
2022-02-28 14:10:48.829 - debug: Redirecting anonymous user to: http://192.168.0.106:8888/o/docs/
2022-02-28 14:10:48.855 192.168.0.106:8888 GET / 302 26.776 ms - 110
2022-02-28 14:10:48.856 - debug: Auth[GET]: id 1 email anon@getgrist.com host 192.168.0.106:8888 path / org docs
2022-02-28 14:10:48.874 192.168.0.106:8888 GET /o/docs/ 200 16.411 ms - 3370
2022-02-28 14:10:48.929 192.168.0.106:8888 GET /v/unknown/jqueryui/themes/smoothness/jquery-ui.css 304 9.057 ms - -
2022-02-28 14:10:48.935 192.168.0.106:8888 GET /v/unknown/bootstrap/dist/css/bootstrap.min.css 304 9.993 ms - -
2022-02-28 14:10:48.936 192.168.0.106:8888 GET /v/unknown/hljs.default.css 304 5.346 ms - -
2022-02-28 14:10:48.937 192.168.0.106:8888 GET /v/unknown/bundle.css 304 4.491 ms - -
2022-02-28 14:10:48.938 192.168.0.106:8888 GET /v/unknown/icons/icons.css 304 4.696 ms - -
2022-02-28 14:10:48.941 192.168.0.106:8888 GET /v/unknown/bootstrap-datepicker/dist/css/bootstrap-datepicker3.min.css 304 9.087 ms - -
2022-02-28 14:10:48.944 192.168.0.106:8888 GET /v/unknown/jquery/dist/jquery.min.js 304 4.263 ms - -
2022-02-28 14:10:48.948 192.168.0.106:8888 GET /v/unknown/jqueryui/jquery-ui.min.js 304 5.666 ms - -
2022-02-28 14:10:48.949 192.168.0.106:8888 GET /v/unknown/bootstrap/dist/js/bootstrap.min.js 304 5.649 ms - -
2022-02-28 14:10:48.950 192.168.0.106:8888 GET /v/unknown/main.bundle.js 304 2.748 ms - -
2022-02-28 14:10:48.951 192.168.0.106:8888 GET /v/unknown/bootstrap-datepicker/dist/js/bootstrap-datepicker.min.js 304 5.218 ms - -
2022-02-28 14:10:48.952 192.168.0.106:8888 GET /v/unknown/browser-check.js 304 1.437 ms - -
2022-02-28 14:10:49.059 192.168.0.106:8888 GET /v/unknown/img/gplaypattern.png 304 1.350 ms - -
2022-02-28 14:11:08.142 192.168.0.106:8888 GET /o/docs/signin?next=http%3A%2F%2F192.168.0.106%3A8888%2Fo%2Fdocs%2F 302 19.932 ms - 110
2022-02-28 14:11:08.147 - debug: Auth[GET]: id 5 email you@example.com host 192.168.0.106:8888 path / org docs
2022-02-28 14:11:08.162 192.168.0.106:8888 GET /o/docs/ 200 10.576 ms - 3370
2022-02-28 14:11:08.222 192.168.0.106:8888 GET /v/unknown/hljs.default.css 304 1.824 ms - -
2022-02-28 14:11:08.223 192.168.0.106:8888 GET /v/unknown/bundle.css 304 1.603 ms - -
2022-02-28 14:11:08.224 192.168.0.106:8888 GET /v/unknown/icons/icons.css 304 1.736 ms - -
2022-02-28 14:11:08.225 192.168.0.106:8888 GET /v/unknown/jqueryui/themes/smoothness/jquery-ui.css 304 5.541 ms - -
2022-02-28 14:11:08.225 192.168.0.106:8888 GET /v/unknown/bootstrap/dist/css/bootstrap.min.css 304 5.314 ms - -
2022-02-28 14:11:08.226 192.168.0.106:8888 GET /v/unknown/bootstrap-datepicker/dist/css/bootstrap-datepicker3.min.css 304 4.832 ms - -
2022-02-28 14:11:08.227 192.168.0.106:8888 GET /v/unknown/jquery/dist/jquery.min.js 304 2.945 ms - -
2022-02-28 14:11:08.228 192.168.0.106:8888 GET /v/unknown/jqueryui/jquery-ui.min.js 304 3.046 ms - -
2022-02-28 14:11:08.229 192.168.0.106:8888 GET /v/unknown/main.bundle.js 304 0.641 ms - -
2022-02-28 14:11:08.230 192.168.0.106:8888 GET /v/unknown/browser-check.js 304 0.751 ms - -
2022-02-28 14:11:08.230 192.168.0.106:8888 GET /v/unknown/bootstrap/dist/js/bootstrap.min.js 304 1.993 ms - -
2022-02-28 14:11:08.230 192.168.0.106:8888 GET /v/unknown/bootstrap-datepicker/dist/js/bootstrap-datepicker.min.js 304 2.021 ms - -
2022-02-28 14:11:08.299 192.168.0.106:8888 GET /v/unknown/img/gplaypattern.png 304 0.493 ms - -
```

**Comment by paulfitz** [comment]
@KucharczykL looking at the logs, does port `8888` mean anything to you in your network setup? I see this line:
```
2022-02-28 14:10:48.827 - debug: Auth[GET]: id 1 email anon@getgrist.com host 192.168.0.106:8888 path / org 
```
where I would have expected:
```
2022-02-28 14:15:28.319 - debug: Auth[GET]: id 1 email anon@getgrist.com host localhost:8484 path / org 
```
If Grist is ultimately being viewed via a different domain or port, it is currently important to set some variables, see e.g. https://github.com/gristlabs/grist-core/issues/70#issuecomment-895168852 - there's an improvement coming that should streamline this.

**Comment by KucharczykL** [comment]
Hi @paulfitz!

`8888` is the port I am exposing it on because `8484` is already in use in my environment, that is I have used the command `docker run -p 8888:8484 -it gristlabs/grist` to run Grist.

This is the first time I have come across the Docker port being different from the host port causing issues (I run ~30 containers).

I have just tried to set the env variable `PORT` to 8888 and that seems to have fixed it.

Thank you for the nudge! :smiley:
